File: 프로그래머스/2/42578. 의상/의상.py
from collections import Counter
def solution(clothes):
    count_gory = Counter(category for name , category in clothes) #카테고리만 세는 카운터
    
    answer = 1
    for  category , cnt in count_gory.items():
        answer *= (cnt+1)
    return answer -1
        
        
    
        

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;
class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        // 1. 종류 별로 옷 개수를 세기 위한 HashMap 생성
        Map<String , Integer> map = new HashMap<>();
        
        for(int i = 0; i < clothes.length; i++){
            String type = clothes[i][1];
            map.put(type , map.getOrDefault(type , 0) + 1);
        }
        
        for(int count : map.values()){
            answer *= (count + 1);
        }

        return answer -1;
    }
}